### PR TITLE
[OBMIN-181] Advertisement_Filter_Page

### PR DESCRIPTION
**Pull request by cloudmenson**

### CODE REVIEW

File: src/main/resources/react/src/components/common/filtration/styles.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+export const BlockStyleSet = css`
+  display: flex;
+  flex-direction: column;
+  gap: 12px;
+  margin-top: 20px;
+`;
+
+export const CategoryFilter = styled.div`
+  ${styleSet}
```
Comment (Wolshebnik) [inline src/main/resources/react/src/components/common/filtration/styles.js:18]: semicolon at the end

File: src/main/resources/react/src/components/common/filtration/styles.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+`;
+
+export const Filter = styled.div`
+  ${styleSet}
```
Comment (Wolshebnik) [inline src/main/resources/react/src/components/common/filtration/styles.js:22]: the same

File: src/main/resources/react/src/routes/routeConstants.js [hunk 3]
```diff
@@ -12,4 +12,5 @@ export const route = {
   prevProductPage: '/product-page',
   oauthSuccess: '/oauth-success',
   noMatch: '*',
+  SearchResultsPage: 'search-results-page',
```
Comment (Wolshebnik) [inline src/main/resources/react/src/routes/routeConstants.js:15]: noMatch should be last

File: src/main/resources/react/src/routes/routes.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ import Login from 'pages/FormRegister/sign-in';
 import SignUp from 'pages/FormRegister/sign-up';
 import ProductPage from 'pages/ProductPage/ProductPage';
 import FourOhFourPage from 'pages/ErrorPages/FourOhFourPage';
+import SearchResultsPage from 'pages/SearchResultsPage';
```
Comment (Wolshebnik) [inline src/main/resources/react/src/routes/routes.js:12]: sort

File: src/main/resources/react/src/pages/UserInfo/tab_pages/my_activity/styles.js [hunk 5]
```diff
@@ -1,21 +1,13 @@
...
 
-export const CardBlock = styled.div`
-  display: grid;
-  grid-template-columns: repeat(3, 1fr);
-  row-gap: 28px;
-  column-gap: 20px;
-`;
-
-export const StyledCardBlock = styled(CardBlock)`
+export const StylizedCardBlock = styled(CardsContainer)`
```
Comment (Wolshebnik) [inline src/main/resources/react/src/pages/UserInfo/tab_pages/my_activity/styles.js:11]: Are you sure that you can do this and that there are no errors in the console?

File: src/main/resources/react/src/components/local/en.js [hunk 6]
```diff
@@ -306,6 +308,15 @@ const en = {
...
   },
+  ageEnum: {
+    YOUNGER_THAN_1: '0',
+    FROM_1_TO_2: '1 - 2',
+    FROM_3_TO_5: '3 - 5',
+    FROM_6_TO_9: '6 - 9',
+    FROM_10_TO_12: '10 - 12',
+    FROM_12_TO_14: '12 - 14',
+    OLDER_THAN_14: '14+',
+  },
```
Comment (Wolshebnik) [inline src/main/resources/react/src/components/local/en.js:319]: it's not needed here
It exists at react > src > config > ENUM

File: src/main/resources/react/src/components/local/ua.js [hunk 7]
```diff
@@ -305,9 +307,18 @@ const ua = {
...
   },
+  ageEnum: {
+    YOUNGER_THAN_1: '0',
+    FROM_1_TO_2: '1 - 2',
+    FROM_3_TO_5: '3 - 5',
+    FROM_6_TO_9: '6 - 9',
+    FROM_10_TO_12: '10 - 12',
+    FROM_12_TO_14: '12 - 14',
+    OLDER_THAN_14: '14+',
+  },
```
Comment (Wolshebnik) [inline src/main/resources/react/src/components/local/ua.js:318]: it's not needed here
It exists at react > src > config > ENUM

File: src/main/resources/react/src/components/local/ua.js [hunk 8]
```diff
@@ -316,6 +327,36 @@ const ua = {
...
+    SIZE_2: '10 см - 16,5 см',
+    SIZE_3: '10,5 см - 17 см',
+    SIZE_4: '11 см - 18 см',
+    SIZE_5: '11,5 см - 19 см',
+    SIZE_6: '12 см - 20 см',
+    SIZE_7: '12,5 см - 21 см',
+    SIZE_8: '13 см - 22 см',
+    SIZE_9: '13,5 см - 23 см',
+    SIZE_10: '14 см - 24 см',
+  },
```
Comment (Wolshebnik) [inline src/main/resources/react/src/components/local/ua.js:349]: I suggest creating two variables and changing sm at depending on lang

File: src/main/resources/react/src/components/local/en.js [hunk 9]
```diff
@@ -317,6 +328,36 @@ const en = {
...
+    SIZE_2: '10 sm - 16,5 sm',
+    SIZE_3: '10,5 sm - 17 sm',
+    SIZE_4: '11 sm - 18 sm',
+    SIZE_5: '11,5 sm - 19 sm',
+    SIZE_6: '12 sm - 20 sm',
+    SIZE_7: '12,5 sm - 21 sm',
+    SIZE_8: '13 sm - 22 sm',
+    SIZE_9: '13,5 sm - 23 sm',
+    SIZE_10: '14 sm - 24 sm',
+  },
```
Comment (Wolshebnik) [inline src/main/resources/react/src/components/local/en.js:350]: I suggest creating two variables and changing sm at depending on lang

File: src/main/resources/react/src/components/common/filtration/select/index.jsx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+import * as Icon from 'assets/icons';
+
+import * as Styles from './styles';
+
+const Select = ({ title, data }) => {
+  const [select, setSelect] = useState(
+    data.map((item) => ({ ...item, isSelected: false }))
+  );
+
```
[the remark this answers is not in the document]
Comment (Wolshebnik) [reply src/main/resources/react/src/components/common/filtration/select/index.jsx:12]: You don't remove space

File: src/main/resources/react/src/components/common/filtration/index.jsx [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+                  type="checkbox"
+                  onChange={setShoesSize}
+                  key={String(item + idx)}
+                  margin="4px 8px 4px 58px"
+                  selectedValues={shoesSize}
+                  text={
+                    lang === 'en'
+                      ? shoesSizeEnum[item]
+                      : shoesSizeEnum[item].replace('cm', 'см')
+                  }
```
Comment (Wolshebnik) [inline src/main/resources/react/src/components/common/filtration/index.jsx:137]: it calls twice, you should think and create a function

**Review by Wolshebnik — APPROVED** [review]

**Review by smilaya — APPROVED** [review]

**Pull request merged**